### PR TITLE
Log fixes

### PR DESCRIPTION
**Pull request by lpusok**
### Checklist

- Requires _PATCH_ [version update](https://semver.org/)

### Context

Logging language fixes.

Resolves: https://bitrise.atlassian.net/browse/STEP-384

### CODE REVIEW

File: devportalservice/errors.go [hunk 1]
```diff
@@ -9,5 +9,5 @@ type NetworkError struct {
 }
 
 func (e NetworkError) Error() string {
-	return fmt.Sprintf("response %d %s", e.Status, e.Body)
+	return fmt.Sprintf("network request failed with status %d, body (%s)", e.Status, e.Body)
```
Comment (godrei) [inline devportalservice/errors.go:12]: i think it does not make sense to include the Body in the message, it is an io.Reader, it's string representation will not be useful i think.

File: main.go [hunk 2]
```diff
@@ -256,22 +256,24 @@ func (cfg Config) validate() error {
...
 
-	if networkErr, ok := err.(devportalservice.NetworkError); ok && networkErr.Status == http.StatusNotFound {
+	if networkErr, ok := err.(devportalservice.NetworkError); ok && networkErr.Status == http.StatusUnauthorized {
 		log.Debugf("")
-		log.Debugf("%s", notConnected)
-	} else {
-		fmt.Println()
-		log.Errorf("Failed to activate Bitrise Apple Developer Portal connection: %s", err)
-		log.Warnf("Read more: https://devcenter.bitrise.io/getting-started/configuring-bitrise-steps-that-require-apple-developer-account-data/")
+		log.Debugf("%s", "Unauthorized to query Connected Apple Developer Portal Account. This likely happens with a public app's PR build by design, to protect secrets.")
```
Comment (godrei) [inline main.go:269]: What about: `This happens, by design, with a public app's PR build, to protect secrets.`

File: main.go [hunk 3]
```diff
@@ -331,7 +334,7 @@ func main() {
 		fail("Could not configure Apple Service authentication: %v", err)
 	}
 	if authConfig.AppleID != nil && authConfig.AppleID.AppSpecificPassword == "" {
-		log.Warnf("If 2FA enabled, Application-specific password is required when using Apple ID authentication.")
+		log.Warnf("If 2FA enabled Apple ID is used, Application-specific password may be required.")
```
Comment (godrei) [inline main.go:337]: always required

File: step.yml [hunk 4]
```diff
@@ -127,54 +107,77 @@ inputs:
         **NOTE:** Only available from XCode 11.0 and above.
```
Comment (godrei) [inline step.yml:107]: Pls remove the NOTE: Xcode version >= 11 is required for App Store deployment.

**Review by godrei — CHANGES_REQUESTED** [review]

File: devportalservice/errors.go (hunk 1 above)
Comment (lpusok) [reply devportalservice/errors.go:12]: it is a string, in this case

File: step.yml (hunk 4 above)
Comment (lpusok) [reply step.yml]: fixed

File: main.go (hunk 3 above)
Comment (lpusok) [reply main.go]: fixed

File: main.go (hunk 2 above)
Comment (lpusok) [reply main.go]: updated

**Review by godrei — APPROVED** [review]

File: step.yml [hunk 5]
```diff
@@ -10,17 +10,19 @@ description: |-
...
-    - For the Apple ID: Fill out the **Apple ID**, **Password**. Optionally, add the **Application specific password** if you wish to bypass two-factor authentication on iTunesConnect.
-    - For the App Store Connect API: Provide your **API Key URL** (for example, https://URL/TO/AuthKey_something.p8 or file:///PATH/TO/AuthKey_something.p8) and the **Issuer ID** inputs. Please note that connecting with the App Store Connect API works with XCode 11.0 and above releases.  
-  1. To identify the app, set either the **App Apple ID** or the **App Bundle ID**. Note that while neither is marked as **Required**, one of the two MUST have a valid value. 
-  1. If you want to immediately submit your app for an iTunesConnect review, set the **Submit for Review?** input to `yes`. Please note that if you do submit the app for review, the Step will be successful only if the submission is accepted by iTunesConnect.
+  1. Set up your connection depending on which authentication method you wish to use:
+      - Use a previously set up Bitrise Apple Developer Connection, set the **Bitrise Apple Developer Connection** to `automatic` (default), `api_key` or `apple_id`.
```
Comment (zoltan-baba) [inline step.yml:20]:
```suggestion
      - Use a previously set up Bitrise Apple Developer connection: set the **Bitrise Apple Developer Connection** to `automatic` (this is the default setting), `api_key` or `apple_id`.
```

File: step.yml [hunk 6]
```diff
@@ -10,17 +10,19 @@ description: |-
...
+      - Provide manual step inputs, either with Apple ID or with the App Store Connet API key. Set the **Bitrise Apple Developer Connection** to `off`. Use only one of the authentication methods.
```
Comment (zoltan-baba) [inline step.yml:21]:
```suggestion
      - Provide manual Step inputs: either with Apple ID or with the App Store Connet API key. Set the **Bitrise Apple Developer Connection** to `off`. Use only one of the authentication methods. 
```

File: step.yml [hunk 7]
```diff
@@ -10,17 +10,19 @@ description: |-
...
-  1. To identify the app, set either the **App Apple ID** or the **App Bundle ID**. Note that while neither is marked as **Required**, one of the two MUST have a valid value. 
-  1. If you want to immediately submit your app for an iTunesConnect review, set the **Submit for Review?** input to `yes`. Please note that if you do submit the app for review, the Step will be successful only if the submission is accepted by iTunesConnect.
+  1. Set up your connection depending on which authentication method you wish to use:
+      - Use a previously set up Bitrise Apple Developer Connection, set the **Bitrise Apple Developer Connection** to `automatic` (default), `api_key` or `apple_id`.
+      - Provide manual step inputs, either with Apple ID or with the App Store Connet API key. Set the **Bitrise Apple Developer Connection** to `off`. Use only one of the authentication methods. 
+          * For the Apple ID: Fill out the **Apple ID: Email**, **Apple ID: Password**. Add the **Apple ID: Application specific password** if you wish to bypass two-factor authentication.
```
Comment (zoltan-baba) [inline step.yml:22]:
```suggestion
          * For the Apple ID: fill out the **Apple ID: Email** and the **Apple ID: Password** inputs. Add the **Apple ID: Application specific password** if you wish to bypass two-factor authentication.
```

File: step.yml [hunk 8]
```diff
@@ -10,17 +10,19 @@ description: |-
...
+          * For the App Store Connect API: Provide your **API Key: URL** (for example, https://URL/TO/AuthKey_something.p8 or file:///PATH/TO/AuthKey_something.p8) and the **API Key: Issuer ID** inputs. Please note that connecting with the App Store Connect API works with XCode 11.0 and above releases.
```
Comment (zoltan-baba) [inline step.yml:23]:
```suggestion
          * For the App Store Connect API: provide your **API Key: URL** (for example, https://URL/TO/AuthKey_something.p8 or file:///PATH/TO/AuthKey_something.p8) and the **API Key: Issuer ID** inputs. Please note that connecting with the App Store Connect API works with XCode 11.0 and above releases.  
```

**Review by zoltan-baba — CHANGES_REQUESTED** [review]

**Pull request merged**